husarion_ugv_hardware_interfaces/src/robot_system/ugv_system.cpp

```diff
@@ -525,8 +526,13 @@ bool UGVSystem::AreVelocityCommandsNearZero()
   return true;
 }
 
-void UGVSystem::MotorsPowerEnable(const bool enable)
+void UGVSystem::EStopTorqueEnable(const bool enable)
 {
+  const bool e_stop = e_stop_->ReadEStopState();
+  if (!e_stop) {
+    throw std::runtime_error("Can't enable/disable torque when E-Stop is not triggered.");
+  }
+
```

Comment:
Can you explain why are we checking this condition instead of simply triggering e-stop?